modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/components/Select/SingleSelect.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayDropDown from '@clayui/drop-down';
+import React, {useState} from 'react';
+
+import {BaseSelect, CustomItem, SelectProps} from './BaseSelect';
+
+import './index.scss';
+
+interface IProps<T extends CustomItem<number | string> = CustomItem>
+	extends SelectProps {
+	onChange: (selected: T) => void;
```

Comment:
Hi @interaminense, I thought it would be better to keep the `onChange` as a mandatory property because the `SingleSelect` can't be functional without it. In the case you commented above (where I pass `() => {}`), there's no need for it to be functional as it's being used just to display a fixed value to the user. But I am worried about the cases where the developer wants the select to work normally, and probably will not pass the `onChange` right away because it is not mandatory, then the select will not work and they will have to investigate until they find out they need to pass the `onChange` for the select to work. What do you think about it?